docs/qml/qml_constructors.md

```diff
@@ -0,0 +1,191 @@
+from qadence import RY
+
+n_qubits = 5
+
+# Custom scaling function
+fm_full = feature_map(
+    n_qubits = n_qubits,
+    support = tuple(reversed(range(n_qubits))), # Reverse the qubit support to run the scaling from bottom to top
+    param = "x", # Change the name of the parameter
+    op = RY, # Change the rotation gate between RX, RY, RZ or PHASE
+    fm_type = BasisSet.CHEBYSHEV,
+    reupload_scaling = ReuploadScaling.EXP,
+    feature_range = (-1.0, 2.0), # Range from which the input data comes from
+    target_range = (1.0, 3.0), # Range the encoder assumes as the natural range
+    multiplier = 5.0 # Extra multiplier, which can also be a Parameter
+)
+
+from qadence.draw import html_string # markdown-exec: hide
+print(html_string(fm_full, size="6,4")) # markdown-exec: hide
+```
+
+## Hardware-efficient ansatz
+
+Ansatze blocks for quantum machine-learning are typically built following the Hardware-Efficient Ansatz formalism (HEA).
+Both fully digital and digital-analog HEAs can easily be built with the `hea` function. By default,
+the digital version is returned:
+
+```python exec="on" source="material-block" html="1" session="ansatz"
+from qadence import hea
+from qadence.draw import display
+
+n_qubits = 3
+depth = 2
+
+ansatz = hea(n_qubits, depth)
+from qadence.draw import html_string # markdown-exec: hide
+print(html_string(ansatz, size="8,4")) # markdown-exec: hide
+```
+
+As seen above, the rotation layers are automatically parameterized, and the prefix `"theta"` can be changed with the `param_prefix` argument.
+
+Furthermore, both the single-qubit rotations and the two-qubit entangler can be customized with the `operations` and `entangler` argument. The operations can be passed as a list of single-qubit rotations, while the entangler should be either `CNOT`, `CZ`, `CRX`, `CRY`, `CRZ` or `CPHASE`.
+
+```python exec="on" source="material-block" html="1" session="ansatz"
+from qadence import RX, RY, CPHASE
+
+ansatz = hea(
+    n_qubits=n_qubits,
+    depth=depth,
+    param_prefix="phi",
+    operations=[RX, RY, RX],
+    entangler=CPHASE
+)
+from qadence.draw import html_string # markdown-exec: hide
+print(html_string(ansatz, size="8,4")) # markdown-exec: hide
+```
+
+Having a truly *hardware-efficient* ansatz means that the entangling operation can be chosen according to each device's native interactions. Besides digital operations, in Qadence it is also possible to build digital-analog HEAs with the entanglement produced by the natural evolution of a set of interacting qubits, as natively implemented in neutral atom devices. As with other digital-analog functions, this can be controlled with the `strategy` argument which can be chosen from the [`Strategy`](../qadence/types.md) enum type. Currently, only `Strategy.DIGITAL` and `Strategy.SDAQC` are available. By default, calling `strategy = Strategy.SDAQC` will use a global entangling Hamiltonian with Ising-like NN interactions and constant interaction strength,
```

Comment:
```suggestion
Having a truly *hardware-efficient* ansatz means that the entangling operation can be chosen according to each device's native interactions. Besides digital operations, in Qadence it is also possible to build digital-analog HEAs with the entanglement produced by the natural evolution of a set of interacting qubits, as natively implemented in neutral atom devices. As with other digital-analog functions, this can be controlled with the `strategy` argument which can be chosen from the [`Strategy`](../qadence/types.md) enum type. Currently, only `Strategy.DIGITAL` and `Strategy.SDAQC` are available. By default, calling `strategy = Strategy.SDAQC` will use a global entangling Hamiltonian with Ising-like $NN$ interactions and constant interaction strength,
```